photutils/geometry/triangular_overlap.pyx

```diff
@@ -0,0 +1,101 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+cimport numpy as np
+
+__all__ = ['triangular_overlap_grid']
+
+DTYPE = np.float64
+ctypedef np.float64_t DTYPE_t
+
+cimport cython
+
+def triangular_overlap_grid(double xmin, double xmax, double ymin,
+                            double ymax, int nx, int ny, double a,
+                            double b, double c, double theta,
+                            int use_exact, int subpixels):
+    """
+    triangular_overlap_grid(xmin, xmax, ymin, ymax, nx, ny,
+                            a, b, c, theta, use_exact, subpixels)
+
+    Area of overlap between a triangle and a pixel grid.
+
+    Parameters
+    ---------
+    xmin, xmax, ymin, ymax : float
+        Extent of the grid in the x and y direction
+    nx, ny : int
+        Grid dimensions
+    a, b, c : float
+        Sides of the triangle
+    theta : float
```

Comment:
For a triangle, would it not make more sense to allow a free specification of the three corners?
I did not quite follow the individual formulas, but if I can specify the triangle with a,b,c, and theta it must be some special case of a triangle (maybe is one angle assume to be 90 deg?)
At the very least, this should be called "EqualsidedTriangle" or "right-angles triangle" (or whatever the assumption is).